### ISSUE TITLE
Compliance/Penalty - External review penalty summary

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-compliance#89]
#### Description:

When an obligation is paid (equivalent value = $0), an operator can navigate their penalty summary and generate penalty invoice (ticket #73) and download penalty payment instructions (same as obligation payment instructions addressed in #135).

#### Acceptance Criteria:
As an operator, given that I've paid my obligation after the payment deadline, I click can 'Continue' on the compliance obligation report page, and then I can view the penalty summary page and navigate to generate an invoice (button at the bottom of page) or to the payment instructions by clicking continue button.

[Figma](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=23398-27&node-type=frame&t=3G4vpjEYNinjvMsl-1&scaling=scale-down&content-scaling=fixed&page-id=14307%3A1663&starting-point-node-id=20107%3A1497&show-proto-sidebar=1)
- [ ] 3 vertical tabs for issuing a penalty: Review Penalty Summary, Download Payment Instructions, Pay Penalty and Track Payment(s)

Review Penalty Summary tab 

- [ ] A header saying 'Penalty'
- [ ] A button to generate the PDF penalty invoice (see #73 for details)
- [ ] An alert message about FAA interest

- [ ] Display fields for overdue payment penalty:
- Penalty status
- Penalty type
- Days late
- Penalty rate
- Accumulated penalty
- Accumulated compounding
- Penalty amount
- FAA interest (as of today)
- Total amount (as of today)

- [ ] A back button to return to compliance report page 
- [ ] Continue button (goes to the Download Payment Instructions screen)

#### Development Checklist:

- [ ] Create automatic penalty task list with Review Penalty Summary, Download Payment Instructions, and Pay Penalty and Track Payment(s) pages
- [ ] Update task list: 1_manageObligationTaskList.ts 
- [ ] Add logic for expanding Compliance Obligation and Automatic Overdue Penalty sections
- [ ] Create Review Penalty Summary page
- [ ] Create RJSF schema for the Review Penalty Summary page
- [ ] Add data fetching from the /compliance-report-versions/{compliance_report_version_id}/automatic-overdue-penalty endpoint
- [ ] Add the Generate Penalty Invoice button
- [ ] Add tests

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By/Blocking
Blocked by #74 
Blocking #90 

#### Notes:

-

**Comment by NicoleGovvy** [comment]
@nanyangpro I've added the suggested fields for the additional obligation penalty here in this ticket.

**Comment by patrickisaac** [comment]
Refinement Notes:
- @nataliepiguliak will add DC

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EWXw04EEsNlKubOoNSpZqB4BiXQzEbhTGO8zqQ5vwW8d4A?e=1FUyYs).

I can verify that:
- [x] When industry user pays an obligation after the deadline, navigations for penalty becomes visible in the task list: Review Penalty Summary, Download Payment Instructions, Pay Penalty and Track Payment(s)
- [x] When user is in the Review Penalty Summary page, user can see:
   - [x] A header saying "Penalty"
   - [x] A button to generate the PDF penalty invoice - functionality will be implemented in #73 
   - [x] An alert message about FAA interest
   - [x] Display fields for overdue payment penalty:
      - [x] Penalty status
      - [x] Penalty type
      - [x] Days late
      - [x] Accumulated penalty
      - [x] Accumulated compounding
      - [x] Penalty amount
      - [x] FAA Interest (As of today)
      - [x] Total Amount (As of today)
   - [x] A `Back` button to return to compliance report page
   - [x] A `Continue` button that goes to the Download Payment Instructions screen - functionality not implemented yet

Wanted to note the following found during testing:
- The `Automatic Overdue Penalty` is only visible when user is in the `Pay Obligation and Track Payment(s)` page - created #274 to tackle that and discuss expected behaviour as well
~- Tried a scenario where user pays the obligation before the payment deadline and I still saw the `Automatic Overdue Penalty` on the side bar, I couldn't see if we already have a ticket to handle scenarios of when this should display. Please let me know if I should create one! cc: @vesselak~ 

Moving to Pending PO for review of test. Thank you!

**Comment by fviduya** [comment]
Moving this back to In Progress. Per @nataliepiguliak, she will add the fix for the scenario I noted in my previous comment:
> Tried a scenario where user pays the obligation before the payment deadline and I still saw the Automatic Overdue Penalty on the side bar.

**Comment by dleard** [comment]
@fviduya @nataliepiguliak The issue of seeing the penalty section when a user pays before the deadline is handled in #271

**Comment by fviduya** [comment]
I have retested the scenario where payment is made before the deadline, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EY7d1uyILqNGlQCSg6zYY3gBVXDt0YCdTJOBTF3jw8V_oA?e=Mg0jQX&nav=eyJoIjoiMjAwMDIwNTY2MyJ9).

I can verify that the `Automatic Overdue Penalty` is not visible once the user gets to the `Pay Obligation and Track Payment(s)` page. 

Additionally, not included in the test document, I ran the same steps done for #271 regarding changing the penalty status in the backend and running the penalty calculation service. This returns an error both when running the penalty calculation, and  when user tries to get on the Review Penalty Summary page. 

<img width="617" height="191" alt="Image" src="https://github.com/user-attachments/assets/622403a0-3bc1-4caf-8cab-95c5e385c3e1" />

The additional test is just to verify that the penalty does not calculate when payment is made before the deadline. In real life, once the obligation has been met before the deadline, they should not accrue penalty (Penalty set to None), therefore the user will not be able to reach the Review Penalty Summary page.

Moving to Pending PO for review of test. Thank you!

**Comment by vesselak** [comment]
Reviewed testing documentation and accepting ticket
